### PR TITLE
Install swr hook

### PR DESCRIPTION
**Pull request by annamarika**

### CODE REVIEW

File: src/components/Form/Form.js [hunk 1]
```diff
@@ -111,6 +111,25 @@ export default function Form({ onAddProduct }) {
...
+								maxLength: 20,
+							})}
+							placeholder="..."
+						/>
+						{errors.name && errors.name.type === 'required' && (
+							<span>please enter your name</span>
+						)}
+						{errors.name && errors.name.type === 'maxLength' && (
+							<span>Please use less than 20 characters</span>
+						)}
```
Comment (StelzenPieper) [inline src/components/Form/Form.js:154]: Very cool custom error messages :)
However if more than 20 characters are entered then spacing on mobile view messes up.
- Create button shown on top of last form input
- Upload button moved off-screen
Maybe fix with final styling?

File: pages/products.js [hunk 2]
```diff
@@ -1,45 +1,26 @@
...
-	};
-
-	const updateProduct = (id, dataObject) => {
-		const updateProduct = products.map(product =>
-			product.id === id ? { ...dataObject, altText: dataObject.title, id: id } : product
-		);
-		setProducts(updateProduct);
-	};
+export default function Products({ fallback }) {
+	console.log(fallback);
```
Comment (StelzenPieper) [inline pages/products.js]: console.log not required anymore(?)

Comment (StelzenPieper) [reply pages/products.js]: nvm, saw it was removed in another commit :)

File: pages/api/products.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { getProducts } from '../../src/services/get-products';
+
+export default async function handler(request, response) {
+	const products = await getProducts();
+	console.log(products);
```
Comment (StelzenPieper) [inline pages/api/products.js:5]: console.log not required anymore(?)

File: src/Components/Form/Form.js [hunk 4]
```diff
@@ -60,10 +65,22 @@ export default function Form({ onAddProduct }) {
...
+			body: JSON.stringify({
+				image: previewImage.secure_url,
+				name: nameValue,
+				title: titleValue,
+				detail: detailValue,
+				email: emailValue,
+			}),
+		});
+
+		console.log(await response.json());
```
Comment (StelzenPieper) [inline src/Components/Form/Form.js:80]: console.log not required anymore(?)

File: src/Components/Product/Product.jsx [hunk 5]
```diff
@@ -40,19 +41,9 @@ export default function Product(props) {
...
-	image,
-	altText,
-	onDeleteProduct,
-	onEnableEditMode,
-}) {
-	const handleDelete = () => {
-		onDeleteProduct(id);
-	};
+function ProductModeShow({ id, name, title, detail, email, image, altText, onEnableEditMode }) {
+	console.log(image);
```
Comment (StelzenPieper) [inline src/Components/Product/Product.jsx:45]: console.log not required anymore(?)

File: src/Components/Product/Product.jsx [hunk 6]
```diff
@@ -72,7 +64,17 @@ function ProductModeShow({
 					</DefaultButton>
 				)}
 				{asPath !== '/products' && (
-					<DefaultButton onClick={handleDelete}>delete</DefaultButton>
+					<DefaultButton
+						onClick={async () => {
+							const response = await fetch('/api/product/' + id, {
+								method: 'DELETE',
+							});
+							console.log(await response.json());
```
Comment (StelzenPieper) [inline src/Components/Product/Product.jsx:72]: console.log not required anymore(?)

File: src/components/Form/Form.js (hunk 1 above)
Comment (annamarika) [reply src/components/Form/Form.js:154]: I will keep that in min in my next US. It's all about clean up.

File: src/Components/lib/database.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const url = process.env.DB_CONNECTION;
+import mongoose from 'mongoose';
+export async function dbConnect() {
+	try {
+		await mongoose.connect(url);
+		console.log('Connect to MongoDB');
```
Comment (StelzenPieper) [inline src/Components/lib/database.js:6]: console.log not required anymore(?) same in line 8?

**Review by StelzenPieper — APPROVED** [review]

**Pull request merged**